crates/handlers/src/views/recovery/finish.rs

```diff
@@ -226,6 +226,18 @@ pub(crate) async fn post(
             );
     }
 
+    if !password_manager.is_password_complex_enough(&form.new_password)? {
+        // TODO This error should be localised,
+        //      but actually this entire form should be pushed down into the
+        //      React frontend so user gets real-time feedback.
```

Comment:
We should indeed make a new `FieldError::Password { entropy: zxcvbn::Entropy }` enum variant? So that in the templates we can extract the entropy score, the guess times, the feedback suggestions and warnings?

This does make me think though that `is_password_complex_enough(&str) -> bool` isn't the right API, but I'm fine with leaving it like that for now

Reply:
I think the goal would be to get all that stuff client-side, because an interactive form will be much nicer than getting that from the backend. Can't see any drawback really?

Reply:
I guess it depends how much we want the javascript-less flows to behave? Anyway, fine for now 👍